Change field_date to range, update views

froboy opened the pull request:
To test
- build site
- Add some events to /admin/structure/taxonomy/manage/event including sub-events
- Visit /admin/structure/views/view/event_sub_events and observe the events show up when you enter the parent vid

Todo
- [x] change date only to date and time
- [x] add bool field to hide time

labbydev commented:
@froboy LMK when this is ready for review, I'd like to get it into the theming branch.

froboy commented:
<img width="935" alt="add_term___midcamp_2017" src="https://user-images.githubusercontent.com/238201/31026251-6020def2-a50b-11e7-8bdd-0f4f36bdc8c4.png">
I've added the boolean field and changed the date field to datetime. This feels stupid and horrible. Yay Drupal.
